js/modules/k6/grpc/client.go

```diff
@@ -353,49 +253,22 @@ func (c *Client) dial(
 	reflect bool,
 	options ...grpc.DialOption,
 ) error {
```

Comment:
This `Client.dial()` code doesn't seem it has anything to do with JavaScript, what is the reason to not move it to `grpcext`?

Reply:
It opens the connection then it routes through reflection if requested from the parameter, I think this way of connecting it's a property of the js module. :thinking:  It could be a good idea to change the name of the method that probably it's not accurate after the refactor because the real dialling part is now delegated to `grpcext`. This method instead is mostly the core part of the `Connect` method.

Reply:
But it seems like the vast majority of the logic related to the reflection is still can be moved to a `grpcext`. 

I belive we can even make a private a `ReflectionClient`. What js nodule needs are the `*descriptorpb.FileDescriptorSet`, so the `grpcext.Conn` can has this method. What do your think?

Reply:
And by the number of the logic, the current `Client.dial` is more like `Client.reflect` than dial :sweat_smile: